fsspec/implementations/reference.py

```diff
@@ -190,10 +190,16 @@ def _process_references1(self, references):
             for pr in products:
                 key = jinja2.Template(gen["key"]).render(**pr, **templates)
                 url = jinja2.Template(gen["url"]).render(**pr, **templates)
-                offset = int(jinja2.Template(gen["offset"]).render(**pr, **templates))
-                length = int(jinja2.Template(gen["length"]).render(**pr, **templates))
-
-                self.references[key] = [url, offset, length]
+                if "offset" in gen and "length" in gen:
```

Comment:
Raise error if only one or the other is present `("offset" in gen) ^ ("length" in gen)` ?